src/System.Web.OData/OData/Query/ODataQueryOptions.cs

```diff
@@ -84,9 +84,16 @@ public ODataQueryOptions(ODataQueryContext context, HttpRequestMessage request)
             BuildQueryOptions(queryParameters);
 
             Validator = ODataQueryValidator.GetODataQueryValidator(context);
+
+            ColumnOrder = new Dictionary<string, int>();
         }
 
         /// <summary>
+        /// Gets Column order.
+        /// </summary>
+        public static Dictionary<string, int> ColumnOrder { get; private set; }
```

Comment:
Why is this static? Do you really want to share the same dictionary for all objects in the model?

Reply:
I think this is more simple way to store column Order.